bears/general/AnnotationBear.py

```diff
@@ -28,104 +27,212 @@ def run(self, filename, file, language: str, coalang_dir: str = None):
-                                       matches_found))
-        # Yield different results for strings and comments
-        strings_found = tuple(filter(lambda arg: arg in unnested_annotations,
-                                     strings_found))
-        comments_found = tuple(filter(lambda arg: arg in unnested_annotations,
-                                      comments_found))
-        yield HiddenResult(self, {'comments': comments_found,
-                                  'strings': strings_found})
+        string_delimiters = dict(lang_dict["string_delimiters"])
+        multiline_string_delimiters = dict(
+            lang_dict["multiline_string_delimiters"])
+        multiline_comment_delimiters = dict(
+            lang_dict["multiline_comment_delimiters"])
+        comment_delimiter = dict(lang_dict["comment_delimiter"])
+        strings = comments = ()
+        try:
+            strings, comments = self.find_annotation_ranges(
+                file,
+                filename,
+                string_delimiters,
+                multiline_string_delimiters,
+                comment_delimiter,
+                multiline_comment_delimiters)
 
-    @staticmethod
-    def find_with_start_end(filename, file, annot):
+        except NoCloseError as e:
+            yield Result(self, str(e), severity=RESULT_SEVERITY.MAJOR,
+                         affected_code=(e.code,))
+
+        content = {"strings": strings, "comments": comments}
+        yield HiddenResult(self, content)
+
+    def find_annotation_ranges(self, file, filename,
+                               string_delimiters, multiline_string_delimiters,
+                               comment_delimiter, multiline_comment_delimiters):
         """
-        Gives all positions of annotations which have a start and end.
+        Find ranges of all annotations.
+
+        :param file:
+            A tuple of strings, with each string being a line in the file.
+        :param filename:
+            The name of the file.
+        :param string_delimiters:
+            A dictionary containing the various ways to  define single-line
+            strings in a language.
+        :param multiline_string_delimiters:
+            A dictionary contatining the various ways to define multi-line
+            strings in a language.
+        :param comment_delimiter:
+            A dictionary contatining the various ways to define single-line
+            comments in a language.
+        :param multiline_comment_delimiters:
 
-        :param filename: Name of the file on which the bear is running.
-        :param file:     Contents of the file in the form of tuple of lines.
-        :param annot:    A dict containing start of annotation as key and end of
-                         annotation as value.
-        :return:         A set of SourceRanges giving the range of annotation.
+            A dictionary contatining the various ways to define multi-line
```

Comment:
here you are actively changing the style of indentation for params in docstrings as mentioned above. Please don't!